### PR TITLE
fix: use auto seal client for consensus

### PR DESCRIPTION
**Pull request by Rjected**
Ref #2689 

This makes sure to share the same client (e.g. headers / bodies client) between the pipeline and consensus engine. Previously, the network client was always used for the consensus engine, even when the auto-seal clients were used for the pipeline. This introduces an `EitherDownloader` type which makes it much easier to share the same client between the pipeline and consensus engine.

### CODE REVIEW

File: bin/reth/src/node/mod.rs [hunk 1]
```diff
@@ -703,6 +708,80 @@ impl Command {
...
 }
 
+/// A sum type to make initializing a consensus engine with different client types easier.
+#[derive(Debug, Clone)]
+pub enum EitherDownloader<A, B> {
+    /// The first downloader variant
+    Left(A),
+    /// The second downloader variant
+    Right(B),
+}
```
Comment (mattsse) [inline bin/reth/src/node/mod.rs:718]: this does not belong here.

this should be part of the downloader crate.

**Review by mattsse — APPROVED** [review]
this is currently the best approach, 
ideally we wouldn't need this, but we'd need to rewrite how the node is configured (we should)

I moved the impl to downloaders

**Comment by gakonst** [comment]
@mattsse what's your ideal long term solution here?

**Comment by mattsse** [comment]
Clean rewrite of the node command with the goal that it's customizable

**Pull request merged**